src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +963,959 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
+                        success &= false;
+                    }
+
+                    // if sub_fund_grp_cd is 'PFCMR' then bldg_cd must be entered
+                    if (StringUtils.isBlank(buildingCode)) {
+                        putFieldError(errorPath, KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_CAMS_SUBFUNDGROUP_WITH_MISSING_BUILDING_CD, subFundGroupCode);
+                        success &= false;
+                    }
+
+                }
+                else {
+                    // if sub_fund_grp_cd is NOT 'PFCMR', campus code should NOT be entered
+                    if (StringUtils.isNotBlank(campusCode)) {
+                        putFieldError(KFSPropertyConstants.ACCOUNT_DESCRIPTION + "." + KFSPropertyConstants.CAMPUS_CODE, KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_NONCAMS_SUBFUNDGROUP_WITH_CAMPUS_CD_FOR_BLDG, subFundGroupCode);
+                        success &= false;
+                    }
+
+                    // if sub_fund_grp_cd is NOT 'PFCMR' then bldg_cd should NOT be entered
+                    if (StringUtils.isNotBlank(buildingCode)) {
+                        putFieldError(KFSPropertyConstants.ACCOUNT_DESCRIPTION + "." + KFSPropertyConstants.BUILDING_CODE, KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_NONCAMS_SUBFUNDGROUP_WITH_BUILDING_CD, subFundGroupCode);
+                        success &= false;
+                    }
+                }
+            }
+        }
+
+        return success;
+    }
+
+    
+    /**
+     * This method checks to see if the contracts and grants fields are filled in or not
+     *
+     * @param account
+     * @param propertyName - property to attach error to
+     * @return false if the contracts and grants fields are blank
+     */
+    protected boolean checkCGFieldNotFilledIn(CuAccountGlobal account, String propertyName) {
+        boolean success = true;
+        Object value = ObjectUtils.getPropertyValue(account, propertyName);
+        if ((value instanceof String && StringUtils.isNotBlank(value.toString())) || (value != null)) {
+            success = false;
+            putFieldError(propertyName, CUKFSKeyConstants.ERROR_DOCUMENT_ACCT_GLB_MAINT_CG_FIELDS_FILLED_FOR_NON_CG_ACCOUNT, new String[] { account.getSubFundGroupCode() });
+        }
+
+        return success;
+    }
+    
+    /**
+     * This method checks to see if the contracts and grants fields are filled in or not
+     *
+     * @param account
+     * @param propertyName - property to attach error to
+     * @return false if the contracts and grants fields are blank
+     */
+    protected boolean checkCGFieldNotFilledIn(CuAccountGlobal account, Account accountGlobalDetail, String propertyName) {
+        boolean success = true;
+        Object value = ObjectUtils.getPropertyValue(accountGlobalDetail, propertyName);
+        if ((value instanceof String && !StringUtils.isBlank(value.toString())) || (value != null)) {
```

Comment:
Another place where !StringUtils.isBlank could be replaced with StringUtils.isNotBlank.